plugins/Tools.cc

```diff
@@ -339,6 +339,91 @@ void HHAnalyzer::matchOfflineLepton(const HLTProducer& hlt, HH::Dilepton& dilept
     }
 }
 
+float HHAnalyzer::getL1TPhiGen(const Lepton& lep) {
+    if (!lep.gen_matched)
+        return getL1TPhiReco(lep);
+    float gen_pt = lep.gen_p4.Pt();
+    float gen_phi = lep.gen_p4.Phi();
+    float gen_theta = lep.gen_p4.Theta();
+    return gen_phi + lep.charge * (1. / gen_pt) * (10.48 - 5.1412 * gen_theta + 0.02308 * gen_theta * gen_theta);
+}
+
+float HHAnalyzer::getL1TPhiReco(const Lepton& lep) {
+    float pt = lep.p4.Pt();
+    float phi = lep.p4.Phi();
+    float theta = lep.p4.Theta();
+    return phi + lep.charge * (1. / pt) * (10.48 - 5.1412 * theta + 0.02308 * theta * theta);
+}
+
+bool HHAnalyzer::sameEndCapGen(const Lepton& lep1, const Lepton& lep2) {
+    if (!lep1.gen_matched || !lep2.gen_matched)
+        return sameEndCapReco(lep1, lep2);
+    return lep1.gen_p4.Eta() * lep2.gen_p4.Eta() > 0 && std::abs(lep1.gen_p4.Eta()) > 1.24 && std::abs(lep2.gen_p4.Eta()) > 1.24;
+}
+
+bool HHAnalyzer::sameEndCapReco(const Lepton& lep1, const Lepton& lep2) {
+    return lep1.p4.Eta() * lep2.p4.Eta() > 0 && std::abs(lep1.p4.Eta()) > 1.24 && std::abs(lep2.p4.Eta()) > 1.24;
+}
+
+float HHAnalyzer::translatePhi(float phi, float translation/*=0*/) {
+    phi += translation; // translate
+    phi = fmod(phi, 2 * M_PI); // put between -2pi, 2pi
+    phi = (phi > 0) ? phi : (2 * M_PI + phi); // put between 0, 2pi
+    return phi;
+}
+
+int HHAnalyzer::getPhiSector(float phi, float start, float end) {
+    for (int i = 0; i < 6; i++) {
+        if (start + i * M_PI / 3 <= phi && phi < end + i * M_PI / 3)
+            return i;
+    }
+    return -1;
```

Comment:
is there a case where this could happen ? Only if you mis-divide 2pi by 60° intervals ?